enumerable.rb

```diff
@@ -0,0 +1,119 @@
+# Enumerable method
+module Enumerable
+  def my_each
+    arr = self
+    if arr.is_a?(Array)
+      (0..(arr.length - 1)).each do |i|
+        yield(arr[i])
+      end
+    end
+    arr
+  end
+
+  def my_each_with_index
+    arr = self
+    (0..(arr.length - 1)).each do |i|
+      yield(arr[i], i)
+    end
+    arr
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result.push(item) if yield(item)
+    end
+    puts result.to_s
+    result
+  end
+
+  def my_all?
+    my_each do |num|
+      if yield(num) == false
+        puts false
+        return false
+      end
+    end
+    puts true
+    true
+  end
+
+  def my_any?
+    my_each do |num|
+      if yield(num) == true
+        puts true
+        return true
+      end
+    end
+    puts false
+    false
+  end
+
+  def my_none?
+    my_each do |num|
+      if yield(num) == true
+        puts false
+        return false
+      end
+    end
+    puts true
+    true
+  end
```

Comment:
- #my_none? returns false if the block ever returns true for all elements in a range
- Tip: think about how none? works, maybe it is just the opposite of any? ?